h2o-genmodel/src/main/java/hex/genmodel/algos/targetencoder/TargetEncoderMojoModel.java

```diff
@@ -0,0 +1,77 @@
+package hex.genmodel.algos.targetencoder;
+
+import hex.genmodel.MojoModel;
+import hex.genmodel.easy.RowData;
+
+import java.util.Iterator;
+import java.util.Map;
+
+public class TargetEncoderMojoModel extends MojoModel implements FeatureTransformerModel{
+
+  public TargetEncoderMojoModel(String[] columns, String[][] domains, String responseName) {
+    super(columns, domains, responseName);
+  }
+
+  public EncodingMaps _targetEncodingMap;
+  public boolean _withBlending;
+  public double _inflectionPoint;
+  public double _smoothing;
+
+  // Could be passed from the model training phase so that we don't have to recompute it here. Or maybe it is fine as it should be computed only once per mojoModel
+  private double _priorMean = -1; 
+
+  @Override
+  public RowData transform(RowData data) {
+    
+    if(_targetEncodingMap != null) {
+      for (Map.Entry<String, EncodingMap> columnToEncodingsMap : _targetEncodingMap.entrySet()) {
+        String columnName = columnToEncodingsMap.getKey();
+        String originalValue = (String) data.get(columnName);
```

Comment:
@mn-mikke In case of target encoding transformation it is expected that columns are **not numerica**l but **enums**.  Regarding explicit casting - I encode my knowledge that the value is not going to be of any but String type and casting is considered to be a faster way ( FYI https://stackoverflow.com/a/676386/1020472 )
Regarding originals of the quoted comment itself.....this is a consequences of inheriting from Map<String, Object>..... it is a payback for shorter version of implementation. I would instead prefer to use composition with two overloaded `put` methods one for `Double` and another for `String` and it would be more clear for users as with composition we are not inheriting `put(String,Object)`.

Reply:
IMHO this issue will go away if the general approach I mentioned in the comment to the PR is implemented. TE MOJO doesn't need to work with RowData.